### PR TITLE
feat(conf) add new bbdo broker stream configurations

### PR DESCRIPTION
**Pull request by a-launois**
## Description

Create two new stream types for broker configuration : "BBDO Server" and "BBDO Client" 

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [x] 22.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by smau-centreon — APPROVED** [review]

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -16971,3 +16971,45 @@ msgstr "Veuillez utiliser l'interface Web pour installer Centreon."
...
+msgstr "Filtrage des évènements par catégories"
+
+msgid "Server address"
+msgstr "Adresse du serveur"
+
+msgid "Server port"
+msgstr "Port de connexion"
+
+msgid "Retry interval (seconds)"
+msgstr "Intervalle entre deux tentatives"
```
Comment (kduret) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Intervalle entre deux tentatives (en secondes)"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -16971,3 +16971,45 @@ msgstr "Veuillez utiliser l'interface Web pour installer Centreon."
...
+
+msgid "Trusted CA's certificate path (optional)"
+msgstr "Chemin du certificat de l’autorité de certification"
```
Comment (kduret) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Chemin du certificat de l’autorité de certification (optionnel)"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -16971,3 +16971,45 @@ msgstr "Veuillez utiliser l'interface Web pour installer Centreon."
...
+
+msgid "Certificate Common Name (optional)"
+msgstr "\"Common Name\" du certificat du serveur"
```
Comment (kduret) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "\"Common Name\" du certificat du serveur (optionnel)"
```

File: www/class/config-generate/broker.class.php [hunk 4]
```diff
@@ -354,6 +355,45 @@ private function generate($poller_id, $localhost)
         $this->writeFile($this->backend_instance->getPath());
     }
 
+    private function cleanBbdoStreams(array $object): array
```
Comment (kduret) [inline www/class/config-generate/broker.class.php]: missing phpdoc

File: www/class/config-generate/broker.class.php [hunk 5]
```diff
@@ -354,6 +355,45 @@ private function generate($poller_id, $localhost)
...
+    {
+        if (isset($object['input'])) {
+            foreach ($object['input'] as $key => $inputCfg) {
+                if ($inputCfg['type'] === 'bbdo_server') {
```
Comment (kduret) [inline www/class/config-generate/broker.class.php]: create a constant for `bbdo_server`

File: www/class/config-generate/broker.class.php [hunk 6]
```diff
@@ -354,6 +355,45 @@ private function generate($poller_id, $localhost)
...
+                    unset($object['input'][$key]['compression']);
+                    unset($object['input'][$key]['retention']);
+
+                    if ($object['input']['encrypt'] === 'no') {
+                        unset($object['input'][$key]['private_key']);
+                        unset($object['input'][$key]['certificate']);
+                    }
+                }
+                if ($inputCfg['type'] === 'bbdo_client') {
```
Comment (kduret) [inline www/class/config-generate/broker.class.php]: create a constant for `bbdo_client`

File: www/class/config-generate/broker.class.php [hunk 7]
```diff
@@ -354,6 +355,45 @@ private function generate($poller_id, $localhost)
...
+                    if ($object['input'][$key]['encrypt'] === 'no') {
+                        unset($object['input'][$key]['ca_certificate']);
+                        unset($object['input'][$key]['ca_name']);
+                    }
+                }
+            }
+        }
+        if (isset($object['output'])) {
+            foreach ($object['output'] as $key => $inputCfg) {
+                if ($inputCfg['type'] === 'bbdo_server' && $object['output'][$key]['encrypt'] === 'no') {
```
Comment (kduret) [inline www/class/config-generate/broker.class.php]: use constant of `bbdo_server`

File: www/class/config-generate/broker.class.php [hunk 8]
```diff
@@ -354,6 +355,45 @@ private function generate($poller_id, $localhost)
...
+                    unset($object['output'][$key]['private_key']);
+                    unset($object['output'][$key]['certificate']);
+                }
+                if ($inputCfg['type'] === 'bbdo_client' && $object['output'][$key]['encrypt'] === 'no') {
```
Comment (kduret) [inline www/class/config-generate/broker.class.php]: use constant of `bbdo_client`

File: www/class/config-generate/broker.class.php (hunk 4 above)
Comment (kduret) [reply www/class/config-generate/broker.class.php]: `$stream` should more appropriate than `$object`

File: www/install/php/Update-22.10.0-beta.1.php [hunk 9]
```diff
@@ -56,3 +62,310 @@
...
+            "isRequired" => 1
+        ],
+        [
+            "fieldname" => 'transport_protocol',
+            "displayname" => 'Transport protocol',
+            "fieldtype" => 'radio',
+            "description" => 'The transport protocol can be either TCP (binary flow over TCP) or gRPC (HTTP2)',
+            "isRequired" => 1,
+            "defaultValue" => 'gRPC',
+            "optionListId" => 13,
```
Comment (kduret) [inline www/install/php/Update-22.10.0-beta.1.php]: you should not use hardcoded ids, cause some modules can add broker configuration and increment ids
(example : https://github.com/centreon/centreon-bam/blob/develop/www/modules/centreon-bam-server/tools/installBrokerConf.php#L90)

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 10]
```diff
@@ -15178,4 +15178,46 @@ msgid "Host severity"
...
+#  msgid "Filter on event categories"
+#  msgstr ""
+
+#  msgid "Server address"
+#  msgstr ""
+
+#  msgid "Server port"
+#  msgstr ""
+
+#  msgid "Retry interval (seconds)"
```
Comment (smau-centreon) [inline lang/es_ES.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
#  msgid "Retry interval (in seconds)"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 11]
```diff
@@ -16971,3 +16971,45 @@ msgstr "Veuillez utiliser l'interface Web pour installer Centreon."
...
+msgid "Filter on event categories"
+msgstr "Filtrage des évènements par catégories"
+
+msgid "Server address"
+msgstr "Adresse du serveur"
+
+msgid "Server port"
+msgstr "Port de connexion"
+
+msgid "Retry interval (seconds)"
```
Comment (smau-centreon) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "Retry interval (in seconds)"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po (hunk 1 above)
Comment (smau-centreon) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Intervalle entre deux tentatives (en secondes)"
```

File: lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 12]
```diff
@@ -15630,3 +15630,45 @@ msgstr "Severidade de host"
...
+#  msgid "Filter on event categories"
+#  msgstr ""
+
+#  msgid "Server address"
+#  msgstr ""
+
+#  msgid "Server port"
+#  msgstr ""
+
+#  msgid "Retry interval (seconds)"
```
Comment (smau-centreon) [inline lang/pt_BR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
#  msgid "Retry interval (in seconds)"
```

File: lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 13]
```diff
@@ -15618,3 +15618,45 @@ msgstr "Severidade de host"
...
+#  msgid "Filter on event categories"
+#  msgstr ""
+
+#  msgid "Server address"
+#  msgstr ""
+
+#  msgid "Server port"
+#  msgstr ""
+
+#  msgid "Retry interval (seconds)"
```
Comment (smau-centreon) [inline lang/pt_PT.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
#  msgid "Retry interval (in seconds)"
```

**Review by smau-centreon — CHANGES_REQUESTED** [review]

File: www/install/php/Update-22.10.0-beta.1.php [hunk 14]
```diff
@@ -20,21 +20,27 @@
...
 require_once __DIR__ . '/../../class/centreonLog.class.php';
-
 $centreonLog = new CentreonLog();
 
 //error specific content
 $versionOfTheUpgrade = 'UPGRADE - 22.10.0-beta.1: ';
 $errorMessage = '';
 
 try {
+    $errorMessage = "Impossible to update 'cb_field' table";
```
Comment (TamazC) [inline www/install/php/Update-22.10.0-beta.1.php:30]: It not linked to this PR, but $errorMessage should be found inside the catch block.
We could log messaes like : start to modify description as info and if the error is thrown log file would contain detailed information about the reason.

**Review by TamazC — DISMISSED** [review]

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 15]
```diff
@@ -17005,11 +17005,11 @@ msgstr "Adresse du serveur"
 msgid "Server port"
 msgstr "Port de connexion"
 
-msgid "Retry interval (seconds)"
-msgstr "Intervalle entre deux tentatives"
+msgid "Retry interval (en seconds)"
```
Comment (smau-centreon) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "Retry interval (in seconds)"
```

**Review by smau-centreon — CHANGES_REQUESTED** [review]

**Review by smau-centreon — DISMISSED** [review]

**Review by smau-centreon — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**